app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -4,6 +4,7 @@ def google_oauth2
     @user = User.from_omniauth(request.env['omniauth.auth'])
 
     if @user.persisted?
+      UserMailer.with(user: @user).welcome_email.deliver_now
```

Comment:
when does this `welcome_email` email will trigger?

Reply:
sir when user sign in then quickly it will trigger

Reply:
we don't need this every time, only the first time we need that